lib/AST/ASTStructuralEquivalence.cpp

```diff
@@ -1210,6 +1210,45 @@ static bool IsStructurallyEquivalent(StructuralEquivalenceContext &Context,
   if (!::IsStructurallyEquivalent(Context, D1->getType(), D2->getType()))
     return false;
 
+  if (D1->getDescribedFunctionTemplate()) {
+    if (D2->getDescribedFunctionTemplate()) {
+      if (!Context.IsStructurallyEquivalent(D1->getDescribedFunctionTemplate(),
```

Comment:
Maybe you could rewrite this as `return ...`and eliminate an if?

Reply:
I don't think its a good idea since this way the function can be extended without modification of this code snippet.